lib/Doctrine/ODM/MongoDB/Aggregation/Stage/AbstractBucket.php

```diff
@@ -26,18 +26,15 @@ abstract class AbstractBucket extends Stage
 {
     private DocumentManager $dm;
 
-    private ClassMetadata $class;
-
     /** @var Bucket\AbstractOutput|null */
     protected $output;
 
     /** @var Expr|array<string, mixed>|string */
     protected $groupBy;
 
-    public function __construct(Builder $builder, DocumentManager $documentManager, ClassMetadata $class)
+    public function __construct(Builder $builder, DocumentManager $documentManager, private ClassMetadata $class)
```

Comment:
```suggestion
    public function __construct(Builder $builder, DocumentManager $dm, private ClassMetadata $class)
```
So the property can be promoted